### PR TITLE
Add solar boiler

### PR DESCRIPTION
**Pull request by Xavier00014**
Works like they did in 1.12

### CODE REVIEW

File: common/src/main/java/com/gregtechceu/gtceu/common/machine/steam/SteamSolarBoiler.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import net.minecraft.world.entity.player.Player;
+
+public class SteamSolarBoiler extends SteamBoilerMachine {
+
+    public SteamSolarBoiler(IMachineBlockEntity holder, boolean isHighPressure, Object... args) {
+        super(holder, isHighPressure, args);
+    }
+
+    @Override
+    public Direction getFrontFacing() { return Direction.UP; }
```
Comment (screret) [inline common/src/main/java/com/gregtechceu/gtceu/common/machine/steam/SteamSolarBoiler.java]: proper formatting please, no one liners.

File: common/src/main/java/com/gregtechceu/gtceu/common/machine/steam/SteamSolarBoiler.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        if (GTUtil.canSeeSunClearly(getLevel(), getPos())){
+            recipeLogic.setStatus(RecipeLogic.Status.WORKING);
+        }else {
+            recipeLogic.setStatus(RecipeLogic.Status.IDLE);
+        }
+        super.updateCurrentTemperature();
+    }
+
+    @Override
+    protected int getCooldownInterval() { return isHighPressure ? 50 : 45; }
```
Comment (screret) [inline common/src/main/java/com/gregtechceu/gtceu/common/machine/steam/SteamSolarBoiler.java]: same here.

File: common/src/main/java/com/gregtechceu/gtceu/common/machine/steam/SteamSolarBoiler.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Override
+    protected int getCoolDownRate() { return 3; }
```
Comment (screret) [inline common/src/main/java/com/gregtechceu/gtceu/common/machine/steam/SteamSolarBoiler.java]: and here.

File: common/src/main/java/com/gregtechceu/gtceu/common/machine/steam/SteamSolarBoiler.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    @Override
+    public ModularUI createUI(Player entityPlayer) {
+        return super.createUI(entityPlayer)
+                .widget(new ProgressWidget(() -> GTUtil.canSeeSunClearly(getLevel(), getPos()) ? 1.0 : 0.0, 114, 44, 20, 20)
+                        .setProgressTexture(GuiTextures.PROGRESS_BAR_SOLAR_STEAM.get(isHighPressure).getSubTexture(0, 0, 1, 0.5),
+                                GuiTextures.PROGRESS_BAR_SOLAR_STEAM.get(isHighPressure).getSubTexture(0, 0.5, 1, 0.5)));
+    }
+
+    @Override
+    protected void randomDisplayTick(RandomSource random, float x, float y, float z) { }
```
Comment (screret) [inline common/src/main/java/com/gregtechceu/gtceu/common/machine/steam/SteamSolarBoiler.java]: aand here.

**Review by screret — DISMISSED** [review]

**Review by mikerooni — APPROVED** [review]

**Pull request merged**